[development] Compilation error on MeshOptimizerComponentTests

**Describe the bug**

Current `development` does not compiles on my environment, due to a warning about a variable not being used:

```
/<path>/o3de_dev/Gems/SceneProcessing/Code/Tests/MeshBuilder/MeshOptimizerComponentTests.cpp:95:17: error: variable 'i' set but not used [-Werror,-Wunused-but-set-variable]
   95 |             int i = 0;
      |                 ^
```

**Steps to reproduce**
Steps to reproduce the behavior:
1. Compile with clang version 20.1.8 following [Engine and Project Distribution](https://www.o3de.org/docs/user-guide/build/distributable-engine/) guide.

**Expected behavior**

O3DE `development` to compile on my machine.

**Actual behavior**

Build error

**Found in Branch**
`development`

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
2565d94be24033717b84a54f360946f6d622165d

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Fedora Linux
 - Version: 42
 - CPU: AMD Ryzen 5 5600G
 - GPU: AMD Ryzen 5 5600G
 - Memory: 16GB

**Additional context**

I already solved this problem, I'm just opening an issue to keep the consistency of the repository. Soon opening the pull request.

--- a/Gems/SceneProcessing/Code/Tests/MeshBuilder/MeshOptimizerComponentTests.cpp
+++ b/Gems/SceneProcessing/Code/Tests/MeshBuilder/MeshOptimizerComponentTests.cpp
@@ -92,12 +92,10 @@ namespace SceneProcessing
 
             auto mesh = AZStd::make_unique<AZ::SceneData::GraphData::MeshData>();
 
-            int i = 0;
             for (const AZ::Vector3& position : planeVertexPositions)
             {
                 mesh->AddPosition(position);
                 mesh->AddNormal(AZ::Vector3::CreateAxisY());
-                ++i;
             }
 
             mesh->AddFace({0, 1, 2}, 0);

